server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/LocalDeliveryTest.java

```diff
@@ -121,6 +121,49 @@ public void mailShouldBeWellDeliveredByDefaultToUserWhenVirtualHostingIsTurnedOf
         verify(messageManager).appendMessage(any(MessageManager.AppendCommand.class), any(MailboxSession.class));
     }
 
+    @Test
+    public void mailDeliveredShouldIncrementMetricCounter() throws Exception {
+        // Given
+        String username = "receiver";
+        MailboxPath inbox = MailboxPath.forUser(username, "INBOX");
+        MessageManager messageManager = mock(MessageManager.class);
+        when(usersRepository.supportVirtualHosting()).thenReturn(false);
+        when(usersRepository.getUser(new MailAddress("receiver@localhost"))).thenReturn(username);
+        when(usersRepository.getUser(new MailAddress(RECEIVER_DOMAIN_COM))).thenReturn(username);
+        when(mailboxManager.getMailbox(eq(inbox), any(MailboxSession.class))).thenReturn(messageManager);
+        when(session.getUser()).thenReturn(User.fromUsername(username));
```

Comment:
This entire class looks like it should be reworked as it massively relies on mocks.

For now I would ignore it - as these tests are really ugly and need a complete rework (not your fault)